apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Span.java

```diff
@@ -31,6 +31,27 @@
 
     private static final Logger logger = LoggerFactory.getLogger(Span.class);
 
+    /**
+     * General type describing this span (eg: 'db', 'ext', 'template', etc)
+     * (Required)
+     */
+    @Nullable
+    private volatile String type;
```

Comment:
Why do the types have to be `volatile`?

Reply:
Basically everything in all reported objects should be volatile because set by one or more threads and read (for serialization) by a different thread.
However, since we are not doing that now, and since those are only populated on creation- I will remove

Reply:
The disruptor ensures visibility on the reporter thread. But for some properties it's important that when a span is created on one thread and ended on another one, visibility is ensured. The `finished` flag is an example for that.